### ISSUE TITLE
Support some default block types

### ISSUE THREAD

**Issue by larowlan** [phenaproxima/drupal-suite#12]
Operating under the assumption that we're likely to have a landing page content-type we probably want some block content types for inline use in layout builder.

Some that feel obvious

* Rich text (like the default basic block, title + wysiwyg)
* Content Card component - takes an entity reference to a node and draws the card title/link/teaser/image from fields on the node
* Manual Card component - fields for title/link field/teaser/image
* Call to action - title, text link
* Hero - title, image

Could ship with two display modes for the cards, e.g horizontal (two-col) and vertical (single col)

**Comment by phenaproxima** [comment]
Definitely like the idea of shipping components like these.

The problem lies in the underlying implementation. Do we want to rely on blocks for this? If we do, I'm not sure the AX is what we'd necessarily want. We could rely on Paragraphs, at least for now, since we explicitly don't have an upgrade path.

The lack of a community-consensus component-based page building system is a huge liability for Drupal, and we do need to address it, but it's unclear what ecosystem (Paragraphs or LB/blocks) we should support in the near-to-medium term.

**Comment by thejimbirch** [comment]
Feels like an add-on recipe from the work that is starting in this initiative: https://www.drupal.org/about/core/blog/working-toward-an-experience-builder

**Comment by Robbt** [comment]
I think this highlights the sort of challenges that Drupal site-builders have experienced for quite some time and that is the choice of which Drupal ecosystem to invest in and what are the potential gotchas that come along with making certain choices. Experience Builder sounds like a great idea and this quote from the article shows the direction they are planning to take - "Based on our research, evolving Layout Builder, and enhancing it with capabilities that exist in Paragraphs today best meets these criteria."

I think one question to answer is what can be built without relying upon paragraphs and what are the drawbacks in terms of usability. Layout Builder is still pretty clunky to use out of the box and while there are a few add-on modules that do improve things (and could possibly be easily configured and installed via recipe), they still leave quite a bit of gaps in terms of your average site builder experience. [Bootstrap Layout Builder](https://www.drupal.org/project/bootstrap_layout_builder) is one of those modules but it needs refinement out of the box as it does things like provide duplication of columns as well as exposing absurd choices such as 11 column layouts. (It exposes the options for 1-12 column sections which is impractical for most people and clutters the UI). It would also require choosing a bootstrap based theme (which in my opinion might not be a bad idea but is another place where finding the best starting theme to build from isn't obvious).

**Comment by phenaproxima** [comment]
Frankly, I kind of like the idea of relying on Paragraphs _for now_, and changing to Experience Builder when it's ready for that.

Starshot has no upgrade or crossgrade paths, and support for those is not part of its concept, so I think it would be viable to change from one system to the other, and maybe fund a migration path in contrib, independent of Starshot itself.

**Comment by Robbt** [comment]
I agree that the project should move forward with as much value as possible but it's probably useful to pay as much attention to Experience Builder as possible. They're discussing what components should be included [here](https://www.drupal.org/project/experience_builder/issues/3446052)

**Comment by larowlan** [comment]
I strongly disagree with recommending paragraphs because of the N revisions issue. I think if we can't agree here we shouldn't chose either and wait for experience builder

**Comment by catch56** [comment]
Sites that install starshot will need to update and maintain their sites over time, once they're on paragraphs, it will be hard for them to get off it. So we should use blocks which is what core currently has for this, and https://www.drupal.org/project/drupal/issues/3440578 when it's ready.

https://www.drupal.org/project/drupal/issues/3365551 will significantly improve the UX for blocks (by not crowding the sidebar as much).

**Comment by phenaproxima** [comment]
>  I think if we can't agree here we shouldn't chose either and wait for experience builder

I think this would be an extremely dangerous strategic mistake for Starshot. By "extremely dangerous", I mean I think it could sink the entire initiative.

It's my opinion that we are kidding ourselves if we think we can really launch Starshot, and have it succeed in the world, without _some_ kind of page building solution that appeals to real-world users. We have to deliver something, and we might have to deliver it before Experience Builder is "ready".

To be clear, I'm not saying we need to go with Paragraphs for any amount of time. But we DO need to have some kind of plan, and _soon_, for what we _will_ go with, and what kind of path forward it will give us. Even if it's not what we'd ideally like, it needs to be significantly better than LB's OOTB experience right now.

To me, this is a **critical** issue.

**Comment by phenaproxima** [comment]
I talked with @larowlan today about this. I think it makes sense for us to ship a Card block type - it's so widely used that it feels like a total no-brainer. Let's at least build the data model, and then in subsequent PRs figure out how to integrate it into page layouts.

The Card block type should have:

- A field that references image or video media.
- A title
- Brief body text
- A link field

All of these fields should be optional.
